pkg/client/kubernetes/applier.go

```diff
@@ -127,95 +127,104 @@ func (a *defaultApplier) deleteObject(ctx context.Context, desired *unstructured
+					_ = unstructured.SetNestedField(newObj.Object, oldClusterIP, "spec", "clusterIP")
+				}
 			}
-		}
 
-		newETP, _, _ := unstructured.NestedString(newObj.Object, "spec", "externalTrafficPolicy")
-		oldETP, _, _ := unstructured.NestedString(oldObj.Object, "spec", "externalTrafficPolicy")
-
-		if oldSvcType == string(corev1.ServiceTypeLoadBalancer) &&
-			newSvcType == string(corev1.ServiceTypeLoadBalancer) &&
-			newETP == string(corev1.ServiceExternalTrafficPolicyTypeLocal) &&
-			oldETP == string(corev1.ServiceExternalTrafficPolicyTypeLocal) {
-			newHealthCheckPort, _ := nestedFloat64OrInt64(newObj.Object, "spec", "healthCheckNodePort")
-			if newHealthCheckPort == 0 {
-				oldHealthCheckPort, _ := nestedFloat64OrInt64(oldObj.Object, "spec", "healthCheckNodePort")
-				_ = unstructured.SetNestedField(newObj.Object, oldHealthCheckPort, "spec", "healthCheckNodePort")
+			newETP, _, _ := unstructured.NestedString(newObj.Object, "spec", "externalTrafficPolicy")
+			oldETP, _, _ := unstructured.NestedString(oldObj.Object, "spec", "externalTrafficPolicy")
+
+			if oldSvcType == string(corev1.ServiceTypeLoadBalancer) &&
+				newSvcType == string(corev1.ServiceTypeLoadBalancer) &&
+				newETP == string(corev1.ServiceExternalTrafficPolicyTypeLocal) &&
+				oldETP == string(corev1.ServiceExternalTrafficPolicyTypeLocal) {
+				newHealthCheckPort, _ := nestedFloat64OrInt64(newObj.Object, "spec", "healthCheckNodePort")
+				if newHealthCheckPort == 0 {
+					oldHealthCheckPort, _ := nestedFloat64OrInt64(oldObj.Object, "spec", "healthCheckNodePort")
+					_ = unstructured.SetNestedField(newObj.Object, oldHealthCheckPort, "spec", "healthCheckNodePort")
+				}
 			}
-		}
 
-	},
-	corev1.SchemeGroupVersion.WithKind("ServiceAccount").GroupKind(): func(newObj, oldObj *unstructured.Unstructured) {
-		// We do not want to overwrite a ServiceAccount's `.secrets[]` list or `.imagePullSecrets[]`.
-		newObj.Object["secrets"] = oldObj.Object["secrets"]
-		newObj.Object["imagePullSecrets"] = oldObj.Object["imagePullSecrets"]
-	},
-	{Group: "autoscaling.k8s.io", Kind: "VerticalPodAutoscaler"}: func(newObj, oldObj *unstructured.Unstructured) {
-		// Never override the status of VPA resources
-		newObj.Object["status"] = oldObj.Object["status"]
-	},
-}
+		},
+		corev1.SchemeGroupVersion.WithKind("ServiceAccount").GroupKind(): func(newObj, oldObj *unstructured.Unstructured) {
+			// We do not want to overwrite a ServiceAccount's `.secrets[]` list or `.imagePullSecrets[]`.
+			newObj.Object["secrets"] = oldObj.Object["secrets"]
+			newObj.Object["imagePullSecrets"] = oldObj.Object["imagePullSecrets"]
+		},
+		{Group: "autoscaling.k8s.io", Kind: "VerticalPodAutoscaler"}: func(newObj, oldObj *unstructured.Unstructured) {
+			// Never override the status of VPA resources
+			newObj.Object["status"] = oldObj.Object["status"]
+		},
+	}
+
+	DeploymentKeepReplicasMergeFunc = func(newObj, oldObj *unstructured.Unstructured) {
```

Comment:
Nit. Do you want to explicitly type `DeploymentKeepReplicasMergeFunc`?
```suggestion
	DeploymentKeepReplicasMergeFunc = MergeFunc(func(newObj, oldObj *unstructured.Unstructured) {
```